### ISSUE TITLE
Photo loaded only after full contact details fetched

### ISSUE THREAD

**Issue by SimonTank** [nextcloud/contacts#477]
Since version 2.1.0 the contact list width is calculated wrong and the contact preview images are not loaded anymore.
![example](https://user-images.githubusercontent.com/1553610/36126063-1cb5a4b0-1057-11e8-9c45-dd82394ec27d.png)

Switching to another contact in list shortly shows the correct width of the list but all contact circles get shifted up.
List should have a fixed width and contact images should be loaded initially.

server version: 13.0.0.14 no additional apps enabled
tested browsers: Firefox 58 and Chrome 64 on Linux Mint

**Comment by macemoneta** [comment]
Seeing the same on my Chromebook, Chrome 64.

**Comment by skjnldsv** [comment]
List width is a duplicate of #453 
Image are displayed on my setup. Can you clear your cache please? :)

**Comment by SimonTank** [comment]
Clearing the cache does not help, event tried with disabled cache and hardreload. Contact images only loads the image by clicking on a contact.
No errors appear in the debug console.

**Comment by skjnldsv** [comment]
Do you have the same problem on every contacts? or only a few?
If so, can you post a vcf card here so we can test?

**Comment by SimonTank** [comment]
![bildschirmfoto vom 2018-02-13 14-34-21](https://user-images.githubusercontent.com/1553610/36152402-4567c694-10cb-11e8-800a-68dfc668d73a.png)

I created a user called Max Mustermann in the Contacts app same problem. Stumbled over the group name Bug #444, not my day. The problem appears on all contacts. No exception.
[0719500f-fad6-4abc-897b-732173b4c949.vcf.gz](https://github.com/nextcloud/contacts/files/1720049/0719500f-fad6-4abc-897b-732173b4c949.vcf.gz)

**Comment by skjnldsv** [comment]
Sooo, is it working for Max Mustermann after all?

**Comment by mnarwold** [comment]
Having the same issue after updating to NC13 and latest version of contacts. Tested on latest chrome and microsoft Edge

**Comment by skjnldsv** [comment]
Can you try in incognito/private mode? I want to check something :)

**Comment by mnarwold** [comment]
Tried in incognito and have the same problem. It initially loads with the
correct view as it is still bringing up details, but as soon as it finishes
loading the detail panel jumps left.

**Comment by skjnldsv** [comment]
@mnarwold OOoh, I thought you meant the image error.
The width error is known and to be addressed in #453  :)

**Comment by mnarwold** [comment]
OK, Thanks, sorry I missed that in your comment above.

**Comment by skjnldsv** [comment]
No worries! Thanks for reporting! :)

**Comment by SimonTank** [comment]
@skjnldsv The image in the list only loads if you select a contact (Max Mustermann), and only for this one contact you selected. No images shown on initial list load.
Just tested it on a other instance of Nextcloud i have access to, its Nextcloud 12.0.5.3 with Contacts 2.1.0, same problem there.

**Comment by skjnldsv** [comment]
Oooh, Okay, I understand better! Yes, this is because we introduced a specific loading to speed up the app. The photo is only loaded when the contact is clicked (though on 13, the shown contacts on the list gets preloaded). I'll suggest you upgrade to 13. :)

**Comment by SimonTank** [comment]
So the contacts app is not preloading any contact images for the list. Unless you click on a contact. And there is no way to show all images on load?
@skjnldsv I'm already using Nextcloud 13.0.0.14 as documented in the main issue comment.

**Comment by skjnldsv** [comment]
@SimonTank Sorry, I read too fast, your latest message stated 12.0.5.3 so I did not read further! 🙈 
My apologies! :)

I misspoke, Indeed, the preloading of images is only done with full contact details load.

**Comment by jzielke-nli** [comment]
Suggesting to add an option for the plugin to allow users to decide whether to load it or not.

**Comment by hast0011** [comment]
Same issue here since an update to nc13. Before everything works just fine so there is no need to develop something new. Just make it as before. The width of contacts is ok since an update of the contacts app but photos will only show up if I click to the contact.

**Comment by Michele64** [comment]
since we are humans we tend to recognize faces much faster than words, so is there any possibility to implement "always show pictures" in the list? every other contact app and every website where it's important to identify people shows profile pictures by default.

**Comment by jakobroehrl** [comment]
Hi @skjnldsv 
I updated to the newest version. 3.1.1
But the problem exists further on one of my servers. 
It tried varies browsers and deactivated and activated the app. 
Do you have an idea how to fix this?
Thanks!

**Comment by skjnldsv** [comment]
@jakobroehrl you will need nextcloud 16 for that. :)
The feature that allow us to check if a contact have a photo or not without downloading the picture itself is available on 16 only :wink:

**Comment by Michele64** [comment]
I just upgraded to 16 and, tadaaah it finally works! Thank you all!

**Comment by skjnldsv** [comment]
@Michele64 on a note from 0 to 10 where 10 is `very snappy` how would you rate the contacts loading now?

**Comment by Michele64** [comment]
Like 7 or 8

**Comment by skjnldsv** [comment]
Thanks! @Michele64 
Any specific points that would require a dedicated attention? :)

**Comment by Michele64** [comment]
nothing so far...
one thing I don't get: how to change the contact picture...

**Comment by skjnldsv** [comment]
@Michele64 click the avatar, top right menu :)

**Comment by m-a-v** [comment]
I had the same problem. Also with your hint I had to look twice. This should be improved. Could it be that the three dots are behind the darkened layer? If they were white like the 'X' users would find it better.

Thanks a lot to all who participated to develop this feature!

**Comment by Michele64** [comment]
There it is!! on my computer it is nearly invisible, see screenshot...
![Bildschirmfoto_2019-05-10_23-48-40](https://user-images.githubusercontent.com/13287876/57558532-558eb480-737e-11e9-88a9-6fa5f777d812.png)

**Comment by skjnldsv** [comment]
@Michele64 Yep, see #1086

**Comment by akoyaxd** [comment]
For me, the contact pictures still don't load in the list. Only of I open them.

Contacts Version 3.1.3, Nextcloud 16, PHP 7.2 on Ubuntu Server 18.04. Tested with Firefox and Chromium.

**Comment by skjnldsv** [comment]
Hi @akoyaxd could you open a new issue? :)
Also please, attach your browser log (1) and network tab (2) if possible

1. Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.
<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

2. Before you do anything, still on the development tools, there is a tab called `network`. Click on it and then the xhr filter. Do your action and screenshot the network log like this:
![capture d ecran_2018-09-27_21-49-46](https://user-images.githubusercontent.com/14975046/46170857-43548000-c29f-11e8-9403-db72713dd9c1.png)

**Comment by jangop** [comment]
I too have the problem that, for a number of my contacts, there contact pictures are not displayed until I explicitly click on the individual contacts. This persists when I change a given contact, i.e. exchange the picture or modify the contact's name.

Here is an excerpt from one such contact ([attached: `john_doe.vcf`](https://github.com/nextcloud/contacts/files/3438992/john_doe.vcf.txt)):

```
BEGIN:VCARD
VERSION:3.0
FN:John Doe
N:Doe;John;;;
ITEM1.X-ABLABEL:
ITEM2.ORG:Università di Pisa
ITEM2.X-ABLABEL:
PHOTO:data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAA0JCgsKCA0LCg
 sODg0PEyAVExISEyccHhcgLikxMC4pLSwzOko+MzZGNywtQFdBRkxOUlNSMj5aYVpQYEpRUk//2
 wBDAQ4ODhMREyYVFSZPNS01T09PT09PT09PT09PT09PT09PT09PT09PT09PT09PT09PT09PT09P
 T09PT09PT09PT0//wgARCAEAAQADAREAAhEBAxEB/8QAGgAAAwEBAQEAAAAAAAAAAAAAAQIDBAA
 FBv/EABgBAQEBAQEAAAAAAAAAAAAAAAABAgME/9oADAMBAAIQAxAAAAFeHPLpolYpujMTOWatBw

[...]

 ORYr+o8konkXo/9/EdhKiy/TtLhNUdS5bbLuUs6i4i1FzDzT/lMIevos1ESEw6wSj6CTL9IQQS3
 SwfPb/v4mYhuY/S1jyJqipg4l3D6GPUFkdxRscQOu4IgwhE0kYwVBBNYfpCH0VEMO0rSP3FZVVV
 qzTFODHJDHmdk3/TBP/Z
UID:50530145-6dbd-483e-abac-fe129cca031c
REV;VALUE=DATE-AND-OR-TIME:2019-07-28T09:03:38Z
END:VCARD



```
